packages/web3-common/src/errors.ts

```diff
@@ -0,0 +1,333 @@
+/* eslint-disable max-classes-per-file */
+
+import {
+	ERR_RESPONSE,
+	ERR_PARAM,
+	ERR_CONN,
+	ERR_CONN_INVALID,
+	ERR_CONN_TIMEOUT,
+	ERR_CONN_NOT_OPEN,
+	ERR_CONN_CLOSE,
+	ERR_CONN_MAX_ATTEMPTS,
+	ERR_CONN_PENDING_REQUESTS,
+	ERR_INVALID_PROVIDER,
+	ERR_INVALID_RESPONSE,
+	ERR_TX_REVERT_INSTRUCTION,
+	ERR_TX_REVERT_TRANSACTION,
+	ERR_TX,
+	ERR_TX_NO_CONTRACT_ADDRESS,
+	ERR_TX_CONTRACT_NOT_STORED,
+	ERR_TX_REVERT_WITHOUT_REASON,
+	ERR_TX_OUT_OF_GAS,
+	ERR_CONTRACT_RESOLVER_MISSING,
+	ERR_CONTRACT_ABI_MISSING,
+	ERR_CONTRACT_REQUIRED_CALLBACK,
+	ERR_CONTRACT_EVENT_NOT_EXISTS,
+	ERR_CONTRACT_RESERVED_EVENT,
+	ERR_CONTRACT_MISSING_DEPLOY_DATA,
+	ERR_CONTRACT_MISSING_ADDRESS,
+	ERR_CONTRACT_MISSING_FROM_ADDRESS,
+} from './constants';
+
+type ConnectionEvent = { code: string; reason: string };
+type Response<T = unknown> = { error?: { message?: string; data?: T } };
+type Receipt = Record<string, unknown>;
+
+export abstract class Web3Error extends Error {
```

Comment:
are you referring here ts Error [interface error](https://github.com/microsoft/TypeScript/blob/main/lib/lib.es5.d.ts#L972) ? , if so then class should implement not extend.

Reply:
No its JS global `Error` object https://developer.mozilla.org/en-US/docs/web/javascript/reference/global_objects/error

Reply:
in that case `public readonly name: string`; is already in [Error](https://developer.mozilla.org/en-US/docs/web/javascript/reference/global_objects/error#instance_properties)

Reply:
Yes its already in the `Error` object but not `readonly`. So why had to redeclare it here.